runbotics/runbotics-orchestrator-ui/src/src-app/views/process/EditProcessDialog.tsx

```diff
@@ -29,71 +33,147 @@ type EditProcessDialogProps = {
     const { translate } = useTranslations();
+    const dispatch = useDispatch();
 
-    const handleSubmit = () => {
-        onAdd(formState);
+    const [formValidationState, setFormValidationState] = useState<FormValidationState>(initialFormValidationState);
+    const [processFormState, setProcessFormState] = useState<IProcess>();
+    const [searchedDatabaseTags, setSearchedDatabaseTags] = useState<Tag[]>([]);
+    const [selectedTagsNames, setSelectedTagsNames] = useState<string[]>([]);
+    const [search, setSearch] = useState<string>('');
+    const debouncedSearch = useDebounce<string>(search.trim(), DEBOUNCE_TIME);
+
+    const handleTagChange = (updatedTags: string[]) => {
+        setSelectedTagsNames(updatedTags);
+
+        const hasTagBeenAdded = updatedTags.length > selectedTagsNames.length;
+        if (hasTagBeenAdded) {
+            const tagToAdd = (searchedDatabaseTags.find(tag => tag.name === updatedTags.at(-1))) ?? { name: updatedTags.at(-1) };
+            setProcessFormState((prevState) => ({
+                ...prevState,
+                tags: [...processFormState.tags, tagToAdd]
+            }));
+        } else {
+            const filteredTags = [...processFormState.tags].filter((tag) => updatedTags.includes(tag.name));
+            setProcessFormState((prevState) => ({
+                ...prevState,
+                tags: filteredTags
+            }));
+        }
+
+        setSearch('');
+    };
+
+    const handleTagSearch = (e: ChangeEvent<HTMLInputElement>) => {
+        const filteredSearch = e.target.value.replaceAll(/[^A-Za-z0-9]/g, '');
+        if (filteredSearch.length <= MAX_TAG_LENGTH) setSearch(filteredSearch);
     };
 
-    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-        setFormState({ ...formState, [e.target.name]: e.target.value });
+    const handleAddByEnter = (e: KeyboardEvent<HTMLInputElement>) => {
+        if(e.key === 'Enter'
+            && search !== ''
+            && selectedTagsNames.length < MAX_NUMBER_OF_TAGS
+            && !selectedTagsNames.includes(search)
+        ) handleTagChange(selectedTagsNames.concat(search));
     };
 
+    const refreshTagList = (searchedValue: string) => {
+        if(searchedValue === '') return;
+
+        dispatch(processActions.getTagsByName(
+            { filter: { contains: { 'name': debouncedSearch } } }
+        ))
+            .then(unwrapResult)
+            .then((tags) => {
+                setSearchedDatabaseTags(tags);
+            });
+    };
+
+    useEffect(() => {
+        refreshTagList(debouncedSearch);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [debouncedSearch]);
+
+    const checkFormFieldsValidation = () => formValidationState.name;
+
+    const getSearchedTagNames = () => {
+        const searchedTagNames = searchedDatabaseTags.map((tag) => tag.name);
+        return (
+            !searchedTagNames.includes(search) && search !== ''
```

Comment:
imo you should change order to `search !== '' && !searchedTagNames.includes(search)`